server/service/src/processors/transfer/requisition/create_response_requisition.rs

```diff
@@ -89,6 +102,30 @@ impl RequisitionTransferProcessor for CreateResponseRequisitionProcessor {
     }
 }
 
+#[derive(Default)]
+struct Auth {
```

Comment:
I changed it to `AuthorisationFields`. Logic for this would be minimal, in this example we didn't have program_id on requisition yet, once it's there we would only need to update one field `approval_status` and just check if the preference is set, thus removing most of the logic and rendering this struct obsolete